python/ml4ir/base/pipeline.py

```diff
@@ -471,8 +454,8 @@ def finish(self, job_status, job_info):
 
         if self.args.file_handler == FileHandlerKey.SPARK:
             # Copy logs and models to HDFS
-            self.file_io.copy_to_hdfs(self.models_dir_local, self.models_dir, overwrite=True)
-            self.file_io.copy_to_hdfs(self.logs_dir_local, self.logs_dir, overwrite=True)
+            self.file_io.copy_to_hdfs(self.models_dir_local, self.models_dir, overwrite=False)
```

Comment:
Do we want to overwrite the files by default?

Reply:
Pros and cons on both ends. Overwriting is risky as we might lose important previous run. Not doing so, we risk discarding the whole run. Best middle ground would be to create a temporary path in HDFS, write it there and communicate the path to the user

For now, I agree we should stick with `overwrite=False`

Reply:
Why do you not want to overwrite by default? I don't get the logic. Do you want your training jobs that have run for hours to fail because there existed something already?

Reply:
I'm fine if we implement the temp path, but it will be a nightmare to track if we have 200 models training in parallel. I don't think we should remove the `overwrite` without an alternative.